onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -1574,6 +1585,56 @@ bool ConcatOpSupportChecker::HasSupportedInputOutputsImpl(
     return false;
   }
 
+  if (IsQuantizedOp(node_unit)) {
+    std::vector<size_t> input_indices(input_size);
+    std::iota(input_indices.begin(), input_indices.end(), 0);
+    if (!IsQuantizedIOSupported(initializers, node_unit, input_indices, params, IOKind::Input)) {
+      return false;
+    }
+
+    if (!IsQuantizedIOSupported(initializers, node_unit, {0}, params, IOKind::Output)) {
+      return false;
+    }
+
+    // Need to verify all the input and output has the same scale and zp for API 28-
+    if (params.android_feature_level < ANEURALNETWORKS_FEATURE_LEVEL_3) {
+      std::vector<float> input_scales(input_size);
+      std::vector<int32_t> input_zps(input_size);
+      for (const auto input_idx : input_indices) {
+        auto status = GetQuantizationScaleAndZeroPoint(
+            initializers, node_unit.Inputs()[input_idx], node_unit.ModelPath(), input_scales[input_idx], input_zps[input_idx]);
+        if (!status.IsOK()) {
+          LOGS_DEFAULT(ERROR) << "Op [" << op_type << "] name [" << op_name
+                              << "] GetQuantizationScaleAndZeroPoint for input_scale/zp failed, message: "
+                              << status.ErrorMessage();
+          return false;
+        }
+      }
```

Comment:
nit: Could we use this for the first input, and HasRequiredScaleAndZeroPoint for all the others and the output?

Reply:
Yes,